src/core/net/dns_client.cpp

```diff
@@ -374,6 +374,49 @@ Error Client::Response::ReadTxtRecord(Section aSection, const Name &aName, Servi
 
 #endif // OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
 
+#if OPENTHREAD_CONFIG_DNS_CLIENT_ARBITRARY_RECORD_QUERY_ENABLE
+
+Error Client::Response::ReadRecordInfo(uint16_t aIndex, RecordInfo &aRecordInfo) const
+{
+    Error          error;
+    uint16_t       offset;
+    ResourceRecord record;
+
+    VerifyOrExit(aIndex < mAnswerRecordCount + mAuthorityRecordCount + mAdditionalRecordCount, error = kErrorNotFound);
```

Comment:
so the Question records are not included in the indexed records, right?

Reply:
No. Not incldued.